src/main/java/org/broadinstitute/hellbender/utils/runtime/StreamingProcessController.java

```diff
@@ -0,0 +1,598 @@
+    }
+
+    /**
+     * Create a controller using the specified settings.
+     *
+     * @param settings                 Settings to be run.
+     * @param promptForSynchronization Prompt to be used as a synchronization point/boundary for retrieving process
+     *                                 output. Blocking calls that retrieve data will block until a prompt-terminated
+     *                                 block is available.
+     */
+    public StreamingProcessController(final ProcessSettings settings, final String promptForSynchronization) {
+        this(settings, promptForSynchronization, false);
+    }
+
+    /**
+     * Create a controller using the specified settings.
+     *
+     * @param settings                 Settings to be run.
+     * @param promptForSynchronization Prompt to be used as a synchronization point/boundary for blocking calls that
+     *                                 retrieve process output.
+     * @param enableJournaling         Turn on process I/O journaling. This records all inter-process communication to a file.
+     *                                 Journaling incurs a performance penalty, and should be used for debugging purposes only.
+     */
+    public StreamingProcessController(
+            final ProcessSettings settings,
+            final String promptForSynchronization,
+            final boolean enableJournaling) {
+        Utils.nonNull(settings, "Process settings are required");
+        this.settings = settings;
+        this.promptForSynchronization = promptForSynchronization;
+
+        if (enableJournaling) {
+            processJournal.enable(settings.getCommandString());
+        }
+    }
+
+    /**
+     * Starts the remote process running based on the setting specified in the constructor.
+     *
+     * @return true if the process has been successfully started
+     */
+    public boolean start() {
+        if (process != null) {
+            throw new IllegalStateException("This controller is already running a process");
+        }
+        process = launchProcess(settings);
+        startListeners();
+        processStdinStream = getProcess().getOutputStream();
+
+        return process.isAlive();
+    }
+
+    /**
+     * Write some input to the remote process, without waiting for output.
+     *
+     * @param line data to be written to the remote process
+     */
+    public void writeProcessInput(final String line) {
+        try {
+            // Its possible that we already have completed futures from output from a previous command that
+            // weren't consumed before we found the prompt at the last synchronization point. If so, drop them
+            // before we issue a new command (since retaining them could confound synchronization on output from
+            // this command), and warn.
+            if (stdErrFuture != null && stdErrFuture.isDone()) {
+                //logger.warn("Dropping stale stderr output: {}", stdErrFuture.get().getBufferString());
```

Comment:
lets remove these commented out lines or make them debug level or something

Reply:
Yes, those both should have been removed.